src/Symfony/Component/Asset/Context/NullContext.php

```diff
@@ -18,6 +18,13 @@
  */
 class NullContext implements ContextInterface
 {
+    public function __construct()
+    {
+        if (__CLASS__ !== get_class($this)) {
```

Comment:
@stof I understand how it works, my point was just *if* you couple to an interface, you *can* fix at infrastructure level. But *if* you are coupled to a `final` class somewhere in your code, you are not able to use proxy manager.

By "fixing at infrastructure level" I mean your suggestion, for example (#20392):

> This would not be hard to implement. We could just add another attribute alongside lazy (or allow passing lazy: className)

@theofidry

> but I don't see what you mean, declaring a service which class is final and registered as lazy will cause the error

It causes the error right now, but it can be fixed with @stof's idea, for instance.